### PR TITLE
ddl: fix unstable test in the TestDomain

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #31966

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_tiflash_api.go [hunk 1]
```diff
@@ -411,8 +411,14 @@ func HandlePlacementRuleRoutine(ctx sessionctx.Context, d *ddl, tableList []TiFl
...
 func (d *ddl) PollTiFlashRoutine() {
+	defer d.wg.Done()
 	pollTiflashContext := NewTiFlashManagementContext()
 	for {
+		select {
+		case <-d.ctx.Done():
+			return
+		case <-time.After(PollTiFlashInterval):
+		}
 		if d.IsTiFlashPollEnabled() {
```
Comment (hawkingrei) [inline ddl/ddl_tiflash_api.go:422]: it has an ```continue``` when failpoint is enabled. it will cause problems.

**Review by wjhuang2016 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by c4pt0r** [comment]
LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**